### PR TITLE
Use Typedefs for addresses in Mizar

### PR DESCRIPTION
**Pull request by akopich**
This introduces Addres type alias.

Test: Unit, Manual
Bug: http://b/242715279

### CODE REVIEW

**Review by vwbaker — APPROVED** [review]

File: src/MizarBase/include/MizarBase/Address.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#define MIZAR_BASE_ADDRESS_H_
+
+#include <functional>
+
+#include "OrbitBase/Typedef.h"
+
+namespace orbit_mizar_base {
+
+struct AddressTag {};
+using Address = orbit_base::Typedef<AddressTag, const uint64_t>;
```
Comment (florian-kuebler) [inline src/MizarBase/include/MizarBase/Address.h:15]: I guess this needs a comment.

Comment (florian-kuebler) [inline src/MizarBase/include/MizarBase/Address.h:15]: The name must be more descriptive. What kind of address is this used for?
Absolute address (from sampling)? -- I guess so.
Or are those relative addresses?

If we introduce strong types here, it is important to take the chance to not mix those kind of addresses.

File: src/MizarBase/include/MizarBase/Address.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+template <typename Action>
+inline void ForFrames(const std::vector<uint64_t>& frames, Action&& action) {
```
Comment (florian-kuebler) [inline src/MizarBase/include/MizarBase/Address.h:18]: Nit: Following other places, I would name this function `ForEachFrame`

File: src/MizarData/include/MizarData/MizarDataProvider.h [hunk 3]
```diff
@@ -12,12 +12,15 @@
...
 #include "ClientData/CaptureData.h"
 #include "ClientData/CaptureDataHolder.h"
+#include "MizarBase/Address.h"
 #include "MizarBase/Time.h"
 
 namespace orbit_mizar_data {
 
 // Handles one of the two datasets Mizar operates on
 class MizarDataProvider : public orbit_client_data::CaptureDataHolder {
+  using Address = ::orbit_mizar_base::Address;
```
Comment (florian-kuebler) [inline src/MizarData/include/MizarData/MizarDataProvider.h:22]: Also adjust the name here.

**Review by florian-kuebler — COMMENTED** [review]
Thanks for the change. Left some comments. Mostly on naming.

File: src/MizarBase/include/MizarBase/Address.h (hunk 1 above)
Comment (beckerhe) [inline src/MizarBase/include/MizarBase/Address.h:15]: What's the purpose of making the inner type `const`? In other words, why does the type not be a regular value (which includes the ability to overwrite it)?

Comment (akopich) [reply src/MizarBase/include/MizarBase/Address.h:15]: > What's the purpose...

So we can't do `(*address)++`. We actually do the same for id-types. 

> why does the type not be a regular value (which includes the ability to overwrite it)

`Address` still supports assignment operator, the const over inner type merely means we turn-off the non-const getters.

Comment (akopich) [reply src/MizarBase/include/MizarBase/Address.h:15]: I see your point. But this is Mizar-specific type (at least, for now). We don't have relative addresses here. Wouldn't it make sense not to rename at least the aliases?

Comment (florian-kuebler) [reply src/MizarBase/include/MizarBase/Address.h:15]: I would be fine with not renaming the alias. But what would be your point?

Comment (akopich) [reply src/MizarBase/include/MizarBase/Address.h:15]: Using a more descriptive name is supposed to resolve ambiguity (absolute vs relative address), which does not exist in Mizar. So the longer type name wouldn't help here.

Comment (akopich) [reply src/MizarBase/include/MizarBase/Address.h:15]: On the second thought, `using Address = ::blah_blah::AbsoluteAddress` looks strange. I'll rename everything.

**Review by florian-kuebler — APPROVED** [review]
LGTM, Thanks

**Pull request merged**